Firmware/firmwareLiBCM/src/powerSave.cpp

```diff
@@ -173,10 +173,12 @@ void powerSave_sleepIfAllowed(void)
 {
     if ((cellBalance_areCellsBalancing()  == NO) /* LiBCM must stay on for safety */                &&
         (gpio_isGridChargerPluggedInNow() == NO) /* LiBCM must stay on for safety */                &&
-        (time_sinceLatestUserInputUSB_get_ms() > PERIOD_TO_DISABLE_SLEEP_AFTER_USB_DATA_RECEIVED_ms) )
-    {
-        powerSave_gotoSleep();
-    }
+        (time_sinceLatestUserInputUSB_get_ms() > PERIOD_TO_DISABLE_SLEEP_AFTER_USB_DATA_RECEIVED_ms))
+	{
+		if (time_sinceLatestKeyOff_ms_get() < PERIOD_TO_DISABLE_SLEEP_AFTER_KEYOFF_MS) { return; } // Not enough time elapsed since latest Key Off, sleep not allowed yet.
```

Comment:
I don't like way this is implemented.  Invert the conditional logic and then just add your conditions to the existing 'if' statement (starting on line#174).  For example, line 176 would become:  `(time_sinceLatestUserInputUSB_get_ms() > PERIOD_TO_DISABLE_SLEEP_AFTER_USB_DATA_RECEIVED_ms) &&`, then line 177 would become `(time_sinceLatestKeyOff_ms_get() > PERIOD_TO_DISABLE_SLEEP_AFTER_KEYOFF_MS)`, then line 178 would become `(time_sinceLatestGridChargerUnplug_get_ms() > PERIOD_TO_DISABLE_SLEEP_AFTER_KEYOFF_MS))`

This way, we don't need the return statements or nested if statements... basically just keeping adding "must be true" conditionals to the existing 'if' statement.